instrumentation/github.com/gin-gonic/gin/otelgin/gintrace.go

```diff
@@ -85,6 +85,17 @@ func Middleware(service string, opts ...Option) gin.HandlerFunc {
 
 		// pass the span through the request context
 		c.Request = c.Request.WithContext(ctx)
```

Comment:
Wouldn't we be fixing this rather more easily with the following:

```suggestion
		c.Request = c.Request.Clone(ctx)
```

[WithContext](https://pkg.go.dev/net/http#Request.WithContext) performs a shallow clone, and therefore doesn't clone the multipart form.
However, [Clone](https://pkg.go.dev/net/http#Request.Clone) performs a deep copy, and makes an explicit clone of [MultipartForm](https://cs.opensource.google/go/go/+/refs/tags/go1.23.4:src/net/http/request.go;l=406).

Reply:
I think this approach could work, but it feels a bit unusual to enforce form parsing in the middleware.

Reply:
Having to force remove multipart files is a bit unusual too.

Reply:
I'm probably more in favor of doing the initialization on the original request as well, explicitly handling clean up logic does seem a bit strange

Reply:
@dmathieu maybe we should discuss this in the SIG since it might have implications in the other instrumentation modules

Reply:
@akats7 Another option is to save `c.Request` to a variable before calling `c.Request.WithContext(ctx)`. Then, in the defer function, if `c.Request.MultipartForm != nil`, we can reattach the `MultipartForm` to the original request. This way, there won’t be any automatic removing or parsing. If other middlewares handle the form correctly, this approach should work. 
In [http/server.go](https://github.com/golang/go/blob/40b3c0e58a0ae8dec4684a009bf3806769e0fc41/src/net/http/server.go#L2102), the `ServeHTTP` method is called first, which invokes the Gin handlers/middlewares, and then `finishRequest` is called, which removes the temporary files.

I haven’t tested yet, but I think it should work. However, I’m considering opening an issue in Go to get their suggestions. Since it’s closely related to the `http` package and the `WithContext` method is widely used by middlewares and frameworks, I think they might be able to suggest best practice for this situation.

Reply:
@akats7 It seems I can’t get any help from the Go team. I’m not in favor of parsing the form in the middleware, as this step should be explicitly done by the user. Parsing large forms can consume resources, and the `ParseMultipartForm` function has a `maxMemory` parameter that should also be decided by the user.

After giving this more thought over the past few days, I believe this is more of a Go design issue. The `WithContext` and `ParseMultipartForm` behavior is implicit, hard to notice, poorly documented, and lacks a clear best practice solution. I think the reattach approach might be better because it’s less aggressive and can likely be safely applied to other OpenTelemetry middlewares facing the same issue. The only downside, compared to `RemoveAll`, is that it relies on other middlewares to also handle `WithContext` and `ParseMultipartForm` correctly. If all middlewares do the right thing, `net/http` should eventually clean up the temporary files automatically.

Reply:
Close to reduce distractions.